node_modules/oae-authentication/tests/test-auth.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2015 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var assert = require('assert');
+
+var RestAPI = require('oae-rest');
+var TestsUtil = require('oae-tests');
+
+describe('Authentication', function() {
+
+    // Rest context that can be used for anonymous requests on the cambridge tenant
+    var anonymousCamRestContext = null;
+    // Rest context that can be used for anonymous requests on the global admin tenant
+    var anonymousGlobalRestContext = null;
+    // Rest context that can be used every time we need to make a request as a tenant admin
+    var camAdminRestContext = null;
+    // Rest context that can be used every time we need to make a request as a tenant admin
+    var gtAdminRestContext = null;
+    // Rest context that can be used every time we need to make a request as a global admin
+    var globalAdminRestContext = null;
+
+    /**
+     * Function that will fill up the tenant admin and anymous rest context
+     */
+    before(function(callback) {
+        // Prepare the contexts with which we'll perform requests
+        anonymousCamRestContext = TestsUtil.createTenantRestContext(global.oaeTests.tenants.cam.host);
+        anonymousGlobalRestContext = TestsUtil.createGlobalRestContext();
+        camAdminRestContext = TestsUtil.createTenantAdminRestContext(global.oaeTests.tenants.cam.host);
+        gtAdminRestContext = TestsUtil.createTenantAdminRestContext(global.oaeTests.tenants.gt.host);
+        globalAdminRestContext = TestsUtil.createGlobalAdminRestContext();
+
+        return callback();
+    });
+
+    describe('Authentication', function(callback) {
+
+        /**
+         * Test that verifies that a user's login ids can be requested
+         */
+        it('verify that only authorized admins can request a user\'s login ids', function(callback) {
+            // Generate a user id
+            var userId = TestsUtil.generateTestUserId();
+
+            // Verify that an error is thrown when a malformed user id was specified
+            RestAPI.Authentication.getUserLoginIds(globalAdminRestContext, userId, function(err, loginIds) {
+                assert.ok(err);
+                assert.equal(err.code, 400);
```

Comment:
Just FYI on a couple things I'm changing here as well that I noticed:

* We use `assert.strictEqual` instead of `assert.equal` now, and we change instances of `assert.equal` as we come across them. There wasn't really any way for you to know that though :P
* I've added a test both for invalid user ids: `'not an id'` and `'g:not:a-user-id'` to test that we both test that it is a valid id and specifically a user id